realm/src/main/java/io/realm/Realm.java

```diff
@@ -308,11 +308,19 @@ private static Realm createAndValidate(RealmConfiguration configuration, boolean
                     throw e;
                 }
             }
+            setupColumnIndices(realm);
 
             return realm;
         }
     }
 
+    private static void setupColumnIndices(Realm realm) {
+        RealmProxyMediator mediator = realm.configuration.getSchemaMediator();
+        for (Class<? extends RealmObject> modelClass : mediator.getModelClasses()) {
```

Comment:
You might want to consider to use a normal for loop as opposed to the iterator, just from a performance standpoint.